### PR TITLE
[WebKit] Update to Xcode 9 beta 1

### PR DESCRIPTION
**Pull request by migueldeicaza**

### CODE REVIEW

File: src/wkwebkit.cs [hunk 1]
```diff
@@ -61,6 +61,76 @@ interface WKBackForwardList {
...
+		[Async]
+		void LookUpContentRuleList (string identifier, Action<WKContentRuleList, NSError> completionHandler);
+	
+		[Export ("removeContentRuleListForIdentifier:completionHandler:")]
+		[Async]
+		void RemoveContentRuleList (string identifier, Action<NSError> completionHandler);
+	
+		[Export ("getAvailableContentRuleListIdentifiers:")]
+		[Async]
+		void GetAvailableContentRuleListIdentifiers (Action<NSString []> callback);
```
Comment (rolfbjarne) [inline src/wkwebkit.cs]: Wouldn't this be better as `Action<string[]>`? Or does the generator not support that (if so, maybe we should add it)?

File: src/wkwebkit.cs [hunk 2]
```diff
@@ -61,6 +61,76 @@ interface WKBackForwardList {
...
+		string Identifier { get; }
+	}
+
+	[Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface WKContentRuleListStore
+	{
+		[Static]
+		[Export ("defaultStore")]
+		WKContentRuleListStore DefaultStore ();
```
Comment (spouliot) [inline src/wkwebkit.cs]: In general those _defaults_ are bound as properties

File: src/wkwebkit.cs [hunk 3]
```diff
@@ -61,6 +61,76 @@ interface WKBackForwardList {
...
+		[Export ("addObserver:")]
+		void AddObserver (WKHttpCookieStoreObserver observer);
+	
+		[Export ("removeObserver:")]
+		void RemoveObserver (WKHttpCookieStoreObserver observer);
+	}
+
+	[Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
```
Comment (spouliot) [inline src/wkwebkit.cs]: No BaseType for protocols (only Model needs one)
Normal inheritance, if needed, can be used to extend another protocol

File: src/wkwebkit.cs [hunk 4]
```diff
@@ -227,6 +297,58 @@ interface WKSecurityOrigin {
...
+		CGRect Rect { get; set; }
+
+		[Export ("snapshotWidth")]
+		NSNumber SnapshotWidth { get; set; }
+	}
+
+#if XAMCORE_2_0
+	[Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
```
Comment (spouliot) [inline src/wkwebkit.cs]: Same

File: src/wkwebkit.cs [hunk 5]
```diff
@@ -227,6 +297,58 @@ interface WKSecurityOrigin {
...
+		void StartUrlSchemeTask (WKWebView webView, WKUrlSchemeTask urlSchemeTask);
+	
+		[Abstract]
+		[Export ("webView:stopURLSchemeTask:")]
+		void StopUrlSchemeTask (WKWebView webView, WKUrlSchemeTask urlSchemeTask);
+	}
+#endif
+	[Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
```
Comment (spouliot) [inline src/wkwebkit.cs]: Same

File: src/wkwebkit.cs (hunk 2 above)
Comment (migueldeicaza) [inline src/wkwebkit.cs]: Thanks!

File: src/wkwebkit.cs (hunk 1 above)
Comment (migueldeicaza) [inline src/wkwebkit.cs]: Agreed, I did not catch this, thanks for the review.

File: src/wkwebkit.cs (hunk 3 above)
Comment (migueldeicaza) [inline src/wkwebkit.cs]: Thanks - I am getting rusty.

File: src/wkwebkit.cs (hunk 4 above)
Comment (migueldeicaza) [inline src/wkwebkit.cs]: Ditto

**Comment by dalexsoto** [comment]
Unrelated failure in monotouch-test 64 bits -> https://bugzilla.xamarin.com/show_bug.cgi?id=57846

**Pull request merged**